src/ocean/io/model/ISuspendableThrottler.d

```diff
@@ -74,16 +76,76 @@ abstract public class ISuspendableThrottler
 
     public void addSuspendable ( ISuspendable s )
     {
-        if ( !this.suspendables.contains(s) )
+        if (!this.suspendables.contains(s))
         {
             this.suspendables ~= s;
+        }
+
+        if (this.suspended_ != s.suspended())
+        {
             if (this.suspended_)
-            {
                 s.suspend();
-            }
+            else
+                s.resume();
         }
     }
 
+    unittest
+    {
+        import ocean.core.Test;
+
+        static class SuspendableThrottler : ISuspendableThrottler
+        {
+            override protected bool suspend ( )
+            {
+                return false;
+            }
+
+            override protected bool resume ( )
+            {
+                return false;
+            }
+        }
+
+        static class Suspendable : ISuspendable
+        {
+            bool suspended_ = false;
+
+            public void suspend ( )
+            {
+                this.suspended_ = true;
+            }
+
+            public void resume ( )
+            {
+                this.suspended_ = false;
+            }
+
+            public bool suspended ( )
+            {
+                return this.suspended_;
+            }
+        }
+
+        auto suspendable_throttler = new SuspendableThrottler;
+        auto suspendable = new Suspendable;
+
+        // add a suspended ISuspendable to a non suspended throttler
+        suspendable.suspend();
+        suspendable_throttler.addSuspendable(suspendable);
+
+        test!("==")(suspendable.suspended(), false,
+            "Suspended suspendable not resumed.");
+
+        suspendable_throttler.suspendAll();
+
+        // add a resumed ISuspendable to a suspended throttler
+        suspendable.resume();
```

Comment:
Actually this line seems superfluous.

Reply:
Ignore my comment. Maybe it's worth having it. It seems easier to understand the test when you set the state explicitly at the beginning :).

Reply:
You wrote this originally, I have only copied it :D